tensorflow/c/env.cc

```diff
@@ -146,6 +147,16 @@ TF_StringStream* TF_GetLocalTempDirectories() {
   return list;
 }
 
+void TF_GetTempFileName(const char* extension, std::string* name,
```

Comment:
We cannot have `std::string` (or any other C++ type) in the type signature for a C API. The C++ types have no guarantee of stability and since the C API is under `extern C`, they are not even mentioned in the mangled name.

We should use `char**` instead and allocate here and have the caller deallocate. We allocate with `malloc`/`calloc`/`strdup` (probably `strdup` is easier) and caller will deallocate with `free`.

Reply:
Done